internal/dcache/rpc/client/functions.go

```diff
@@ -1295,6 +1296,200 @@ func GetMVSize(ctx context.Context, targetNodeID string, req *models.GetMVSizeRe
+					// then blobfuse2 process or the node has restarted causing those connections to fail
+					// with broken pipe or connection closed/reset errors, so first time around we don't
+					// mark the node as negative, but if retrying also fails with similar error, it means the
+					// blobfuse2 process is still down so we mark it as negative.
+					//
+					cp.deleteAllRPCClients(client, i == 1 /* confirmedBadNode */, false /* isClientClosed */)
+					if i == 1 {
+						break
+					}
+					err1 := cp.waitForNodeClientPoolToDelete(client.nodeID, client.nodeIDInt)
+					if err1 != nil {
+						break
+					}
+					// Continue with newly created client.
+					continue
+				} else if rpc.IsTimedOut(err) {
+					cp.deleteAllRPCClients(client, true /* confirmedBadNode */, false /* isClientClosed */)
+					break
+				}
+
+				// Fall through to release the RPC client.
+				resp = nil
+			} else {
+				//
+				// The RPC call to the target node succeeded. If the node or the RV is marked negative or iffy,
+				// clear it now.
+				//
+				cp.removeNegativeNode(targetNodeID)
+			}
+
+			// Release RPC client back to the pool.
+			err1 := cp.releaseRPCClient(client)
+			if err1 != nil {
+				log.Err("rpc_client::GetLogs: Failed to release RPC client for node %s %v: %v",
+					targetNodeID, reqStr, err1)
+				// Assert, but not fail the GetLogs call.
+				common.Assert(false, err1)
+			}
+		}
+
+		//
+		// If the GetLogs RPC failed, the log tar file was written partially.
+		// So, we delete this file.
+		//
+		if err != nil {
+			common.Assert(resp == nil)
+
+			// We create the output file when chunkIndex is 0.
+			if chunkIndex == 0 {
+				common.Assert(fh == nil, reqStr)
+				common.Assert(len(outPath) == 0, reqStr)
+			} else {
+				common.Assert(fh != nil, reqStr)
+				common.Assert(len(outPath) > 0, reqStr)
+
+				// Delete the output file.
+				err1 := os.Remove(outPath)
+				_ = err1
+				common.Assert(err1 == nil, err1, outPath, reqStr)
+			}
+
+			return nil, err
+		}
+
+		common.Assert(resp != nil, reqStr)
+		common.Assert(len(resp.Data) > 0, reqStr, resp.ChunkIndex, resp.IsLast, resp.TarName, resp.TotalSize)
+
+		//
+		// For local node, the first GetLogs RPC call for chunkIndex=0 will create the log tarball
+		// in /tmp/<tarName> path. So, we don't make RPC calls again for next chunks as we can directly
+		// copy the local file from /tmp/<tarName> to <outDir>/<tarName>.
+		//
+		if targetNodeID == rpc.GetMyNodeUUID() {
+			common.Assert(chunkIndex == 0, reqStr)
+
+			srcPath := filepath.Join(os.TempDir(), resp.TarName)
+			outPath = filepath.Join(outDir, resp.TarName)
+
+			log.Debug("rpc_client::GetLogs: Copying local log tar file from %s -> %s", srcPath, outPath)
+			err = copyFile(srcPath, outPath)
+			if err != nil {
+				log.Err("rpc_client::GetLogs: Failed to copy log tar file in local node %v [%v]", err)
+				return nil, err
+			}
+
+			break
+		}
```

Comment:
no need to special case this. The advantage is miniscule and it ties the server and client other than the rpc call semantics.